api/src/test/java/org/openmrs/module/fhir2/api/translators/impl/EncounterTranslatorImplTest.java

```diff
@@ -250,11 +247,33 @@ public void toOpenMrsType_shouldTranslateParticipantToEncounterProvider() {
 		Patient patient = new Patient();
 		patient.setUuid(PATIENT_UUID);
 		when(patientReferenceTranslator.toOpenmrsType(patientRef)).thenReturn(patient);
+		when(encounterService.getEncounterRoleByName("Unknown")).thenReturn(new EncounterRole());
+		when(participantTranslator.toOpenmrsType(ArgumentMatchers.any(), ArgumentMatchers.any())).thenReturn(new EncounterProvider());
 		org.openmrs.Encounter result = encounterTranslator.toOpenmrsType(fhirEncounter);
 		
 		assertThat(result.getEncounterProviders(), not(empty()));
 		assertThat(result.getEncounterProviders().size(), equalTo(1));
 	}
+
+	@Test(expected = IllegalStateException.class)
+	public void toFhirResource_shouldThrowExceptionWhenUknownRoleIsNull() {
+		List<Encounter.EncounterParticipantComponent> participantComponents = new ArrayList<>();
+		Encounter.EncounterParticipantComponent participantComponent = new Encounter.EncounterParticipantComponent();
+		Reference practitionerRef = new Reference();
+		practitionerRef.setReference(PRACTITIONER_URI);
+		participantComponent.setIndividual(practitionerRef);
+		participantComponents.add(participantComponent);
+		fhirEncounter.setParticipant(participantComponents);
+		EncounterProvider encounterProvider = new EncounterProvider();
+		Provider provider = new Provider();
+		provider.setUuid(PRACTITIONER_UUID);
+		encounterProvider.setProvider(provider);
+		Patient patient = new Patient();
+		patient.setUuid(PATIENT_UUID);
+		when(patientReferenceTranslator.toOpenmrsType(patientRef)).thenReturn(patient);
+		when(participantTranslator.toOpenmrsType(ArgumentMatchers.any(), ArgumentMatchers.any())).thenReturn(new EncounterProvider());
+		encounterTranslator.toOpenmrsType(fhirEncounter);
```

Comment:
```suggestion

		encounterTranslator.toOpenmrsType(fhirEncounter);
```

Reply:
Prefer using `assertThrows()` with a lambda over `expected` in the test annotation.